### PR TITLE
[cmake] Turn on root7 if CMAKE_CXX_STANDARD allows it

### LINKED ISSUE #6406: root7 is turned off by default even if the default C++ standard of the compiler is C++14 or above
## Describe the bug

With [ROOT-10692](https://sft.its.cern.ch/jira/browse/ROOT-10692) fixed, ROOT now, by default, uses the default C++ standard of the compiler rather than always using C++11. However, due to how our cmake logic is structured, root7 is still turned off by default, even if the default C++ standard used by the compiler was detected to be C++14 or higher.

## Expected behavior

With a compiler that defaults to `-std=C++14` or above, a vanilla `cmake path/to/root` should have root7 turned on.

## Additional context

I think the root cause is that, at the following lines in our main `CMakeLists.txt`, we _first_ include `RootBuildOptions` (which sets root7 to OFF by default because it doesn't detect a high-enough C++ standard) and _then_ we include `CheckCompiler`, which sets our default `CMAKE_CXX_STANDARD` to the compiler default.

https://github.com/root-project/root/blob/33458dc53621cf3e685a302a4bf899108aed0d0c/CMakeLists.txt#L128-L134

Moving `include(CheckCompiler)` above `include(RootBuildOptions)` fixes this issue but breaks Windows, because some cmake variable that `CheckCompiler` needs in the case of windows were defined earlier.

### PR DESCRIPTION
**Pull request by eguiraud**
This fixes #6406.

### CODE REVIEW

**Comment by eguiraud** [comment]
@phsft-bot build

**Comment by eguiraud** [comment]
@phsft-bot build please

**Comment by eguiraud** [comment]
@phsft-bot build

File: cmake/modules/CheckCompiler.cmake [hunk 1]
```diff
@@ -198,7 +198,7 @@ if(CMAKE_SYSTEM_NAME MATCHES Linux)
 elseif(APPLE)
   include(SetUpMacOS)
 elseif(WIN32)
-  include(SetupWindows)
```
Comment (oshadura) [inline cmake/modules/CheckCompiler.cmake:201]: 0_0 wow!

**Review by oshadura — APPROVED** [review]
LGTM, all changes are reasonable (I am sure windows build will be green, but let's wait before merging)

File: cmake/modules/CheckCompiler.cmake (hunk 1 above)
Comment (eguiraud) [reply cmake/modules/CheckCompiler.cmake:201]: cmake is mind-boggling. variable names are case-sensitive but commands and some other things are not.

**Comment by oshadura** [comment]
@phsft-bot build just on windows10/cxx14

**Comment by eguiraud** [comment]
@phsft-bot build just on windows10/cxx14

**Pull request merged**